instant-xml/src/impls.rs

```diff
@@ -25,3 +28,149 @@ impl<'xml> FromXml<'xml> for bool {
         }
     }
 }
+
+// Serializer
+
+macro_rules! to_xml_for_number {
+    ($typ:ty) => {
+        impl ToXml for $typ {
+            fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+            where
+                W: fmt::Write,
+            {
+                match serializer.consume_field_context() {
+                    Some(field_context) => {
+                        match field_context.attribute {
+                            Some(FieldAttribute::Attribute) => {
+                                serializer.add_attribute_value(&self.to_string());
+                            }
+                            _ => {
+                                serializer.add_open_tag(&field_context)?;
+                                write!(serializer.output, "{}", &self)?;
+                                serializer.add_close_tag(field_context)?;
+                            }
+                        }
+                        Ok(())
+                    }
+                    None => Err(Error::UnexpectedValue),
+                }
+            }
+        }
+    };
+}
+
+to_xml_for_number!(i8);
+to_xml_for_number!(i16);
+to_xml_for_number!(i32);
+to_xml_for_number!(i64);
+to_xml_for_number!(isize);
+to_xml_for_number!(u8);
+to_xml_for_number!(u16);
+to_xml_for_number!(u32);
+to_xml_for_number!(u64);
+to_xml_for_number!(usize);
+to_xml_for_number!(f32);
+to_xml_for_number!(f64);
+
+impl ToXml for bool {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        let value = match self {
+            true => "true",
+            false => "false",
+        };
+
+        match serializer.consume_field_context() {
+            Some(field_context) => {
+                match field_context.attribute {
+                    Some(FieldAttribute::Attribute) => {
+                        serializer.add_attribute_value(value);
+                    }
+                    _ => {
+                        serializer.add_open_tag(&field_context)?;
+                        serializer.output.write_str(value)?;
+                        serializer.add_close_tag(field_context)?;
+                    }
+                }
+                Ok(())
+            }
+            None => Err(Error::UnexpectedValue),
+        }
+    }
+}
+
+impl ToXml for String {
+    fn serialize<W>(&self, serializer: &mut Serializer<W>) -> Result<(), Error>
+    where
+        W: fmt::Write,
+    {
+        match serializer.consume_field_context() {
```

Comment:
We should be able to just call the `&str` impl from here instead of redoing it.

Reply:
done